swift_scality_backend/diskfile.py

```diff
@@ -325,7 +342,123 @@ def delete(self, timestamp):
                           `DiskFile` class in Swift. This `delete` method is
                           called externally only by the `ObjectController`
         """
-        self._filesystem.del_object(self._name)
+        self.client_collection.get_write_client().del_object(self._name)
+
+
+class NoClientAvailable(RuntimeError):
+    '''Exception raised when no client with alive endpoints is available'''
+
+
+class ClientCollection(object):
```

Comment:
This file is already big. I'd consider moving this class either in its own file or in http_utils.py.